core/packages/contracts/contracts/Vault.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: transfer ownership from deployer to SovereignTreasury
+// This contract actually holds Ether balances for each sovereignID.
+contract Vault is Ownable {
+    using Address for address payable;
+
+    event Deposited(bytes32 indexed sovereignID, uint256 amount);
+    event Withdrawn(bytes32 indexed sovereignID, address recipient, uint256 amount);
+
+    // Mapping of sovereignID to balance
+    mapping(bytes32 => uint256) private _balances;
```

Comment:
No underscore prefixes please. That's really for larger, more complicated contracts where local variables can shadow global variables.

```suggestion
    mapping(bytes32 => uint256) private balances;
```